### ISSUE TITLE
Global Header menu: Use icons for the mobile menu toggle buttons

### ISSUE THREAD

**Issue by marko-srb** [WordPress/wporg-mu-plugins#463]
Can we change the 'Menu' in mobile to the cheveron? As in design... 

Screenshot for reference.

<img width="669" alt="Screen Shot 2023-09-28 at 21 21 16" src="https://github.com/WordPress/wporg-showcase-2022/assets/111751264/04412065-b6d8-4687-b2a9-f61cfa1dbbc4">

It's good to consider the same edit/look for Plugins:
[https://github.com/WordPress/wporg-mu-plugins/](https://www.google.com/url?q=https://github.com/WordPress/wporg-mu-plugins/&sa=D&source=docs&ust=1695929438279329&usg=AOvVaw3QBYey04L0Wvmo9a4MMWd0)

**Comment by ryelle** [comment]
The "Submit a site" link is part of the navigation, so it is also hidden in the mobile dropdown - I don't think we can (or should) separate it out like that.

> It's good to consider the same edit/look for Plugins:

Are you saying that this should be done across all redesigned sites?

**Comment by marko-srb** [comment]
Disregard the link (can/should go under the arrow). 

Oh, let's do it for Showcase only. I thought it was only present on the plugins (live sites) at the moment. 

—

But generally I think yes, all redesigned sites should have this use-case with cheveron. @jasmussen can help me out, but I believe all redesigned mobile versions should go with cheveron instead of text "Menu". 
That could be a separate issue, for sure, with further discussion. 

Reference.
<img width="601" alt="Screen Shot 2023-09-28 at 21 33 57" src="https://github.com/WordPress/wporg-showcase-2022/assets/111751264/13276b2b-ba16-45b0-9160-f839da8e44ae">

When @fcoveram is back from vacation we can discuss more. Although if I recall it was approved design...

**Comment by ryelle** [comment]
It is present on live (redesigned) sites, see https://wordpress.org/about/requirements/, https://wordpress.org/download/counter/, https://wordpress.org/documentation/ etc.

> But generally I think yes, all redesigned sites should have this use-case with cheveron

I don't want us to make this change twice, and I don't think it makes sense for one site (showcase) to diverge without it being very intentional. So I'd recommend either changing everywhere, or holding off entirely if it needs more discussion.

For what it's worth, "Menu" as the label was proposed/decided [in this discussion starting here](https://github.com/WordPress/wporg-mu-plugins/issues/465).

**Comment by jasmussen** [comment]
I missed that initial conversation around replacing the chevron with the word "Menu", and the close with the "X". I'd argue there's nuance there around legibility, especially with long titles. So I agree, we should go back to having the chevron to open, and an X to close. CC: @fcoveram for when he has a chance to catch up.

Apologies for the back and forth there.

**Comment by ryelle** [comment]
I've moved the issue to wporg-mu-plugins so this can be implemented on all redesigned sites. Unfortunately I think there will still need to be a change in every child theme to update the navigation block's `hasIcon`, combined with the CSS and workarounds to get the navigation block to show a custom icon (the only core icons are the two hamburger variants).

I'm unclear about the timing here, though. Is this required for the showcase launch? Should it wait for @fcoveram's feedback too?

For now, I've left it in the showcase project, but as blocked.

**Comment by marko-srb** [comment]
Ideally it is done for Showcase before the launch, and discussed with @fcoveram before we do it on all other places. If that's not the option, then the whole thing can wait. 

Thanks.

**Comment by ryelle** [comment]
Okay, I'm going to take it out of the Showcase project then.

**Comment by ndiego** [comment]
For reference, there is a discussion in Core about allowing custom icons: https://github.com/WordPress/gutenberg/issues/37930. 

That said, in the meantime, we can use the `WP_HTML_Tag_Processor`. The following code should work but probably could be cleaned up a bit, and the icons might need some design modifications. 

You will need to set the Navigation block to show the **three-bar hamburger icon** on mobile. The two-bar icon does not use `path` and, therefore won't work. Then, this code will convert the icon to chevrons. 

```php
function customize_navigation_block_icon( $block_content, $block ) {

    if ( $block['blockName'] === 'core/navigation' ) {

        $tag_processor = new WP_HTML_Tag_Processor( $block_content );

        if ( 
            $tag_processor->next_tag( array( 
                'tag_name' => 'div', 
                'class_name' => 'wp-block-wporg-local-navigation-bar' 
            ) )
        ) {
            if ( 
                $tag_processor->next_tag( array( 
                    'tag_name' => 'button', 
                    'class_name' => 'wp-block-navigation__responsive-container-open' 
                ) ) &&
                $tag_processor->next_tag( 'path' )
            ) {
                $tag_processor->set_attribute( 'd', 'M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z' );
            }
    
            if ( 
                $tag_processor->next_tag( array( 
                    'tag_name' => 'button', 
                    'class_name' => 'wp-block-navigation__responsive-container-close' 
                ) ) &&
                $tag_processor->next_tag( 'path' )
            ) {
                $tag_processor->set_attribute( 'd', 'M6.5 12.4L12 8l5.5 4.4-.9 1.2L12 10l-4.5 3.6-1-1.2z' );
            }
    
            return $tag_processor->get_updated_html();
        }
    }

    return $block_content;
}
add_filter('render_block', 'customize_navigation_block_icon', 10, 2);
```

**Comment by jasmussen** [comment]
Just wanted to clarify that from design, let's definitely go with the icons, not the labels here. I understand if this can be blocked or challenged technically, in which case it isn't the most urgent in the world, but just noting for the record so I can unblock it from design.

**Comment by ryelle** [comment]
It's not technically difficult (@ndiego's solution looks good, or CSS like I said), I was just saying that it's not a "flip a switch" change. It was blocked on the design confirmation that this should happen for _all_ sites, so I've moved it to "to do" now.

**Comment by fcoveram** [comment]
This is definitely my mistake. Sorry for missing the "Menu" label decision and not updating the component. I second @jasmussen's point on going with the icon.

I will update the component to reflect the change in all mockups. Thanks and sorry for the back and forth.

**Comment by ndiego** [comment]
@renintw and @adamwoodnz, this is not a requirement for the Showcase v2 launch, but this should now be unblocked. We will move forward with the icon instead of the "Menu". Let us know if you need anything else. Thanks!

**Comment by adamwoodnz** [comment]
[Developer](https://developer.wordpress.org/redesign-test/) is updated, I'll roll out the rest today.

**Comment by adamwoodnz** [comment]
✅ Rolled out to Showcase, Documentation and Main too.
